### ISSUE TITLE
Painfully slow / unusable Electron apps (Disord, Slack, Battle.net / Blizzard Launcher)

### ISSUE THREAD

**Issue by levinium** [electron/electron#12988]
I know that this forum was probably not designed for this sort of a bug report, but I don't know where else to turn.

I have Discord, Slack, and the Battle.net (Blizzard Launcher) app all the latest versions installed on my Windows 8.1 machine, and all of them have been running extremely painfully slow for about 2 or so weeks now (clicking on things, typing, animations coming out of the app are all lagged, and it seems like it's all running at somewhere between .3-1 fps .. I also can't use Discord for voice chat at all due to this bug, it just won't register me talking, but I can hear other people talking) .. they all used to run great, but suddenly (and I'm not sure why) this started happening.

I discovered that the common technology between Slack and Discord is Electron! So now I'm finally on to finding what might be going on here, but I'm still no closer to a solution.

If I run Slack or Discord in a web browser (google chrome) they work just fine, but their Electron apps are completely unusable (along with the Battle.net app also running slow in the same way, which I have to guess is also somehow Electron based / related).

My computer isn't chugging away or anything, looking at process explorer doesn't reveal anything unusual about the CPU or memory usage of these apps. I've run sfc /scannow to no avail, I've also (as a blizzard post suggested) tried changing my internet settings (make sure there's no proxy, and check or uncheck the automatically detect settings option in Internet Options) to no avail.

Any ideas? I've googled this extensively and nothing seems to be on the right track here, but it's got to be Electron related since it's only happening specifically for these electron apps like Slack and Discord.

Is there something Electron related on my system that may be corrupted that I can uninstall/reinstall/repair, or an electron specific cache I can clear? Or anything I can potentially try to address this? These electron apps are basically unusable because of this bug.

Any ideas would be helpful.

Thanks all!

EDIT / SOLUTION:

OK so I had a hunch and tried something and I think I found the cause of the problem.. VMware? I had VMware installed because I was testing some things on a virtual machine, and I thought that perhaps the way it sets up the network card to share to its virtual machines may have somehow interfered with Electron.. so I uninstalled VMware and instantly all Electron apps are now again running smoothly. Not sure what exactly happened here, and apparently I can't both have VMware installed and use Electron based apps, but the problem is now SOLVED!

Why would VMware cause all Electron apps to lag in this way? I'm guessing something to do with the network interfaces it creates / modifies to get its virtual machines to have internet.. but this does seem like some sort of a bug with Electron + VMware, so I guess that it's good to have here?

**Comment by levinium** [comment]
OK so I had a hunch and tried something and I think I found the cause of the problem.. VMware? I had VMware installed because I was testing some things on a virtual machine, and I thought that perhaps the way it sets up the network card to share to its virtual machines may have somehow interfered with Electron.. so I uninstalled VMware and instantly all Electron apps are now again running smoothly. Not sure what exactly happened here, and apparently I can't both have VMware installed and use Electron based apps, but the problem is now SOLVED!

**Comment by sedwards2009** [comment]
For the kind of slow down you describe I would also explore interactions between VMware, Electron and the GPU/graphics subsystem in Windows. I sounds like those Electron apps couldn't use the GPU properly.

Disclaimer: not an expert on GPUs, Windows or VMware.

**Comment by codebytere** [comment]
Closing on perceived resolution above.

**Comment by levinium** [comment]
problem came back after I restarted =\

**Comment by brenca** [comment]
TBH if other people can confirm this then it's probably a bug, but so far it seems something unique to your setup is causing this, and it might not be a bug in electron.

**Comment by sqlwhale** [comment]
I'm having a similar issue on a VMware based virtual desktop. Visual Studio Code, SQL Ops Studio, Mattermost, and GitKraken. I have plenty of resources and the delay between action and response is crazy. Sometimes a 5 second delay from key press to response. I can see animations in the applications frame by frame. It's super frustrating. I restart seems to help but very quickly it goes right back to happening.

**Comment by Demotron** [comment]
Hi guys! I have the same issue. I use VMware as my working environment and I can't uninstall it completely. So it's not the solution at all. I'm having problems with VS Code and Slack. The timeout is up to several seconds between input and response.

**Comment by Petschko** [comment]
Battle.net uses Electron? That explains why it always hang, oh god, I hope this framework dies out like flash did...

Thanks for the new Flash! We don't need our memory for better things, so thanks for using it, free memory is wasted memory right?

**Comment by KevinXuzio** [comment]
Just decided I'd post my experience here as this was the first location I found when googling the issue:

I tried uninstalling VMWare as above suggested, that was not the issue in my case - however the issue for me turned out to be Windows Defender for some reason - not sure what was exactly causing it however I disabled windows defender and the issue disappeared. I would suggest using some alternative protection to defender

Hopefully this helps someone

**Comment by EnderDev** [comment]
I've even gone as far as resetting my PC and still no luck.

**Comment by marksalpeter** [comment]
Experiencing the same issue on a windows 8.1 machine right now. We badly need a definitive solution. Why is this closed?

**Comment by catzcod** [comment]
This is not a joke! I have no VMWare, but Windows 8.1 and have the same interface issues with MS Teams, Signal, both electron. Why is this thread closed?!

**Comment by Stokoe0990** [comment]
Linux user here, apps themselves aren't slow for me.

But my whole system starts to bog-down if I've got in excess of 2 Electron-based apps running.

Any combination of VSCode, MailSpring, Insomnia, Terminus or Hyper.js will cause my system to start noticeably stuttering, the more running, the more it slows down.

I'd expect this with 4GB of RAM, but not 16GB.

**Comment by highfeed** [comment]
I have no VMWare but I have the same interface issues.

**Comment by pcfreak30** [comment]
Linux user as well. Running Manjaro19. I have both slack with 13 teams and discord running and getting lags and non-responsiveness for up to 10 seconds before it catches up. I have an AMD ryzen 7, 8 core CPU, and 32 GB of ram, AND SSD RAID, so the hardware and software are definitely not the issues.

I don't use VMware, I do use VirtualBox, but it's not running. I have a dedicated Nvidia GPU as well.

**Comment by kiview** [comment]
I am assuming this happens e.g. in Slack on Linux, because hardware acceleration is not working correctly?

**Comment by Stokoe0990** [comment]
> I am assuming this happens e.g. in Slack on Linux, because hardware acceleration is not working correctly?

This would have been my first assumption, too.
Alas, my issue seems to have gone away since I tweaked file watching things to miss dependency directories such as node_modules or vendor etc.

**Comment by catzcod** [comment]
On Windows system with Nvidia GPU, app freezes when it is using Nvidia instead of Intel graphics. Need to select integrated Intel card for this problematic app and it will run as usual again.

**Comment by TheRedXD** [comment]
Guys, I have had the same issue. I even overclocked my CPU, it didn't help that much but it still did something. Though it's not that painful. I had Visual Studio Code, and it was running pretty fine but now its just extremely painfully slow. I realized it uses electron, and I had issues with electron. I also have a Virtual Machine program, called VirtualBox, and that might be the issue. If you're uninstalling VMWare or any other kind of VM app, then make sure all of it's adapters are removed as well.

**Comment by TheRedXD** [comment]
My friend is familiar with electron, so he coded an electron app, but it was also very slow. The website is pretty fast though.
Here's the code:
```js
const electron = require('electron');
const app = electron.app;
app.on('ready', async() => {
    var mainWindow = new electron.BrowserWindow({
        "title": "PrismClient"
    });
    mainWindow.setMenuBarVisibility(false);
    mainWindow.loadURL("https://prismclient--redprogaming.repl.co/login.html");
});```

**Comment by TheRedXD** [comment]
I had issues with node.js running slowly, so maybe that could relate to the issue. Mac's handle it good cause node runs on them pretty fast.

**Comment by Stokoe0990** [comment]
> My friend is familiar with electron, so he coded an electron app, but it was also very slow. The website is pretty fast though.
> Here's the code:
> [...]


That's just Google Chrome with extra steps.

Node is just a resource hog, always has been. And it runs just as badly on OSX as any other OS.

**Comment by maxim-allen** [comment]
I can confirm that the problem is valid in July 2020, at least to me.
Hardware
1st machine  - 24 GB RAM, i7-8565U, Windows 10 Pro
2md machine - 12 GB RAM, Phenom II 905e, Windows 7 Pro

**Comment by SpaceDux** [comment]
For me.
I have my main workstation (Ubuntu 20.04, AMD Ryzen 5 2700x, AMD RX500XT, 16gb RAM), my app runs like a dream (unpackaged)
However when I'm doing remote work on my laptop (i5, 8gb ram), it runs painfully slow, and there's just nothing I can do to get it to run like it does on my workstation.
Resources are fine on my laptop, only thing running is Atom & then my app+cmd.

This has only just started to happen. And I'm out of ideas to speed it up, I'm assuming it's a Windows / Node+Electron issue, I've tried to keep my code as clean and minimal as possible, disabled hardware acceleration, but still no joy. 

Haven't tested a packaged version of my app, but I imagine the issue will persist. I have another electron app that is packaged for windows, and works excellently. Very strange issue.

**Comment by w-henderson** [comment]
I'm having the same problem... my PC runs fine but my laptop really lags on Discord and my personal electron apps. I thought it was something to do with switchable graphics or GPU drivers but updating them didn't fix it. My laptop is an Acer Nitro 5 (2017 AMD version).

**Comment by hussaa1** [comment]
I am working on an Angular 8 app packaged with package-win to Electron Node JS. This app is based on 10 tabs. I have had several people test this app for me and most people have not reported much bad experience except one person with i5 processor on Windows 10. I do have VirtualBox installed myself but I don't run it while testing this App and I don't see much performance issues except on couple of tabs that contain just over 1000 option radio boxes that are processed through backend Typescript. I have personally tried on several platforms i.e. Windows 7 SP3 x86, Windows 10 Pro, x86, x64 with i3 and i7 processors. Only these two tabs show slight performance bottleneck. i.e. the page may take about 1 to 3 seconds on load. Then works as normal. One person on the first time run of this app for me have complained that the tab with high number of option radio boxes took almost 40 seconds to load the first time only. After that it now runs as normal with the expected delays. One person's app doesn't even run on his i5 processor machine and the container shows only white screen. 

It is very strange that Electron app can have so many diverse permeances on various machines and we are unable to get some help? 

Here's a peak of the radio option buttons tab.

![image](https://user-images.githubusercontent.com/5120276/99815319-07956800-2b42-11eb-9ee6-0e2a8de3d558.png)

**Comment by rhysce** [comment]
For me the issue (Debian, 5 second typing lag for both slack app and browser version, no issue for discord in browser) was resolved by disabling fractional scaling.

**Comment by hyped0001** [comment]
Was this ever solved? My electron app is only experiencing this issue on Mac's with Intel processor. M1 Macs and Windows PCs the app runs fine. Any thoughts?

**Comment by soldierlytomcat** [comment]
I am also having problems with this too. Searched and tried out many solutions out there with little success. All electron based apps (Discord, Vscode, Steam-for some parts of the interface) and browsers (Chrome & Edge but Brave for some reason) I have on my computer lags.
Turning off hardware acceleration for my browsers worked, until I tried using Figma on my browser. Turns out figma does need hardware acceleration to function properly, or otherwise is subject to lags and rendering issues.
I went looking again till I came across a fix, which modifies some flags in chromium based browsers. This seems to work as Figma now works fine...buuuuuut any page that's rendering-heavy like Figma's protoyping seems to fail again.
I have found no fix for other apps, and at my wits end here.
